src/main/java/org/greekleanersinc/servicetemplate/service/TemplateServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package org.greekleanersinc.servicetemplate.service;
+
+import io.grpc.stub.StreamObserver;
+import lombok.extern.slf4j.Slf4j;
+import net.devh.boot.grpc.server.service.GrpcService;
+import org.greekleanersinc.servicetemplate.ServiceData;
+import org.greekleanersinc.servicetemplate.Request;
+import org.greekleanersinc.servicetemplate.Response;
+import org.greekleanersinc.servicetemplate.TemplateServiceGrpc;
+import org.greekleanersinc.servicetemplate.model.TemplateData;
+import org.greekleanersinc.servicetemplate.repository.TemplateRepository;
+
+@GrpcService
+@Slf4j
+public class TemplateServiceImpl extends TemplateServiceGrpc.TemplateServiceImplBase {
+    private final TemplateRepository templateRepository;
+
+    public TemplateServiceImpl(TemplateRepository templateRepository) {
+        this.templateRepository = templateRepository;
+    }
+
+    @Override
+    public void findTemplate(Request request,
+                             StreamObserver<Response> responseObserver) {
+        Long id = request.getId();
+        log.info("Finding template by id: {}", id);
+
+        var template = templateRepository.findById(id);
+        Response response = Response
+                .newBuilder()
+                .setResponseData(convertToProto(template))
+                .build();
+        responseObserver.onNext(response);
+        responseObserver.onCompleted();
```

Comment:
Medium. Should we have an additional layer where we would accumulate GRPC-related stuff? Not sure if we want to have this boilerplate at service level.

Reply:
Major: Also, this layer can be used by the client module which would allow us to make client independent of implementation, if we introduce Service-level interfaces.

So, cross-module dependency schema can look like this:
1 Dependency: client -> grpc with service-interfaces
2 Dependency: service-impl -> service interfaces

Reply:
You're absolutely right, thx! it's fixed